src/volue/mesh/tests/test_model.py

```diff
@@ -0,0 +1,444 @@
+    
+    parent_object_id = session2.get_object(parent_object_path).id
+    session2.delete_object(parent_object_id)
+    
+    # commit the deletion of parent
+    session2.commit()
+    
+    # commit on creation of child should fail
+    with pytest.raises(grpc.RpcError, match="Missing owner"):
+        session1.commit()  
+   
+    session1.close()
+    session2.close()
+    
+        
+@pytest.mark.database
+def test_two_sessions_adding_child_after_removing_parent2(connection):
+     
+    root_object_ownership_relation_attribute_parent = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef"
+    root_object_ownership_relation_attribute_child = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef/TestPlantParent.SimpleOwnershipAtt"
+        
+    session1 = connection.create_session()
+    session1.open()
+  
+    session2 = connection.create_session()
+    session2.open()
+   
+    #first create the parent objectd
+    session1.create_object(root_object_ownership_relation_attribute_parent,"TestPlantParent")
+    session1.commit()
+    
+    #delete parent object    
+    parent_object_path = f"{root_object_ownership_relation_attribute_parent}/{'TestPlantParent'}"
+    
+    parent_object_id = session2.get_object(parent_object_path).id
+    session2.delete_object(parent_object_id)
+    
+    # commit the deletion of parent
+    session2.commit()
+    
+    # commit on creation of child should fail
+    
+    with pytest.raises(grpc.RpcError, match="Owner of the object not found"):
+        session1.create_object(root_object_ownership_relation_attribute_child,"TestPlantChild")
+    
+    #commit on session1 is not needed because create should fail
+    
+    session1.close()
+    session2.close()
+
+    
+@pytest.mark.database
+def test_two_sessions_removing_same_object(connection):
+    
+    root_object_ownership_relation_attribute = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef"
+    #root_object_ownership_relation_attribute_child = "Model/SimpleThermalTestModel/ThermalComponent.ThermalPowerToPlantRef/TestPlantParent.SimpleOwnershipAtt"
+        
+    session1 = connection.create_session()
+    session1.open()
+  
+    session2 = connection.create_session()
+    session2.open()
+   
+    #first create the object to be deleted
+    session1.create_object(root_object_ownership_relation_attribute,"TestPlantToRemove")
+    session1.commit()
+    
+    #deleting the object in session1
+    new_object_path = f"{root_object_ownership_relation_attribute}/{'TestPlantToRemove'}"
+    
+    new_object_id_session1 = session1.get_object(new_object_path).id
+    session1.delete_object(new_object_id_session1)
+    session1.commit()
+    
+    #deleting the same object in session2 should fail
+    
+    with pytest.raises(grpc.RpcError, match="Object not found"):
+        new_object_id_session2 = session2.get_object(new_object_path).id
+        session2.delete_object(new_object_id_session2)
```

Comment:
If the `get_object` raises an error then we will never execute the `detele_object`. Add separate:

```suggestion
    with pytest.raises(grpc.RpcError, match="Object not found"):
        session2.get_object(new_object_path)

    with pytest.raises(grpc.RpcError, match="Object not found"):
        session2.delete_object(new_object_path)
```